pkg/monitoring/metrics/virt-controller/metrics.go

```diff
@@ -42,31 +67,36 @@ var (
 		vmSnapshotMetrics,
 	}
 
-	vmInformer                    cache.SharedIndexInformer
-	vmiInformer                   cache.SharedIndexInformer
-	persistentVolumeClaimInformer cache.SharedIndexInformer
-	vmiMigrationInformer          cache.SharedIndexInformer
-	kvPodInformer                 cache.SharedIndexInformer
-	clusterConfig                 *virtconfig.ClusterConfig
-	instancetypeMethods           *instancetype.InstancetypeMethods
+	informers     *Informers
+	stores        *Stores
+	clusterConfig *virtconfig.ClusterConfig
+	vmApplier     vmApplyHandler
 )
 
 func SetupMetrics(
-	vm cache.SharedIndexInformer,
-	vmi cache.SharedIndexInformer,
-	pvc cache.SharedIndexInformer,
-	vmiMigration cache.SharedIndexInformer,
-	pod cache.SharedIndexInformer,
+	metricsInformers *Informers,
+	metricsStores *Stores,
 	virtClusterConfig *virtconfig.ClusterConfig,
-	methods *instancetype.InstancetypeMethods,
+	clientset kubecli.KubevirtClient,
 ) error {
-	vmInformer = vm
-	vmiInformer = vmi
-	persistentVolumeClaimInformer = pvc
-	vmiMigrationInformer = vmiMigration
-	kvPodInformer = pod
+	informers = metricsInformers
+	stores = metricsStores
```

Comment:
Initialize with empty structs if nil was passed in `metricsInformers` / `metricsStores`?